File: accommodation-booking/server/src/Services/Search/dto/accommodation_dto.py

class AccommodationDTO:
    def __init__(self, accommodation, num_guests, start_date, end_date):
        self.id = accommodation.id
        self.hostId = accommodation.host_id
        self.name = accommodation.name
        self.description = accommodation.description
        self.location = accommodation.location.to_son().to_dict()
        self.minGuests = accommodation.min_guests
        self.maxGuests = accommodation.max_guests
        self.photo = accommodation.photo
        self.benefits = [benefit.to_son().to_dict() for benefit in accommodation.benefits]
        self.priceType = accommodation.price_type
        self.price = accommodation.calculate_price(num_guests, start_date, end_date)

    def list_to_DTO(accommodation_list, num_guests, start_date, end_date):
        return [AccommodationDTO(accommodation, num_guests, start_date, end_date).__dict__ for accommodation in accommodation_list]

File: accommodation-booking/server/src/Services/Search/Search.API/DTO/AccommodationDTO.cs
using Search.API.Models;

namespace Search.API.DTO
{
    public class AccommodationDTO
    {
        public Guid Id { get; private set; }
        public Guid HostId { get; private set; }
        public string Name { get; private set; }
        public string Description { get; private set; }
        public Address Location { get; private set; }
        public int MinGuests { get; private set; }
        public int MaxGuests { get; private set; }
        public string Photo { get; private set; }
        public List<Benefit> Benefits { get; private set; }
        public PriceType PriceType { get; private set; }
        public float Price { get; private set; }

        public AccommodationDTO(Accommodation accommodation, int numOfGuests, DateTime start, DateTime end)
        {
            Id = accommodation.Id;
            HostId = accommodation.HostId;
            Name = accommodation.Name;
            Description = accommodation.Description;
            Location = accommodation.Location;
            MinGuests = accommodation.MinGuests;
            MaxGuests = accommodation.MaxGuests;
            Photo = accommodation.Photo;
            Benefits = accommodation.Benefits;    
            PriceType = accommodation.PriceType;
            Price = accommodation.CalculatePrice(numOfGuests, start, end);
        }
    }
}
